sdk/all/src/test/java/io/opentelemetry/sdk/OpenTelemetrySdkTest.java

```diff
@@ -76,10 +76,38 @@ void testReconfigure() {
             .setClock(clock)
             .setResource(resource)
             .build();
-    assertThat(openTelemetry.getTracerProvider()).isEqualTo(tracerProvider);
+    assertThat(((ObfuscatedTracerProvider) openTelemetry.getTracerProvider()).unobfuscate())
+        .isEqualTo(tracerProvider);
     assertThat(openTelemetry.getMeterProvider()).isEqualTo(meterProvider);
     assertThat(openTelemetry.getPropagators()).isEqualTo(propagators);
     assertThat(openTelemetry.getResource()).isEqualTo(resource);
     assertThat(openTelemetry.getClock()).isEqualTo(clock);
   }
+
+  @Test
+  void testReconfigure_justClockAndResource() {
+    Resource resource = Resource.create(Attributes.builder().put("cat", "meow").build());
+    OpenTelemetrySdk openTelemetry =
+        OpenTelemetrySdk.builder().setClock(clock).setResource(resource).build();
+    TracerProvider unobfuscatedTracerProvider =
+        ((ObfuscatedTracerProvider) openTelemetry.getTracerProvider()).unobfuscate();
+
+    assertThat(unobfuscatedTracerProvider).isInstanceOf(TracerSdkProvider.class);
+    // this is not a great way to test this...
```

Comment:
```suggestion
    // Since TracerProvider is in a different package, the only alternative to this reflective
    // approach would be to make the fields public for testing which is worse than this.
```